dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6325,6 +6325,25 @@ bool isMCSTable(TABLE* table_ptr)
         return false;
 }
 
+bool isForeignTableUpdate(THD* thd)
+{
+    LEX* lex = thd->lex;
+
+    if (lex->sql_command != SQLCOM_UPDATE_MULTI)
+        return false;
+
+    Item_field* item;
+    List_iterator_fast<Item> field_it(lex->first_select_lex()->item_list);
+
+    while ((item = (Item_field*) field_it++))
+    {
+        if (item->field && item->field->table && !isMCSTable(item->field->table))
```

Comment:
Does this check work with views and subqueries?

Reply:
This check does work if a view is getting updated. I have added a test case to cover this.

Target of the UPDATE operation cannot be a subquery. A subquery can however be the source. I have added a test case for this.